### PR TITLE
Support configuring election timeout range

### PR DESCRIPTION
**Pull request by BusyJay**
Sometimes, a node is expected to take less chance to become a leader due to reasons like poor performance. We can make it happen by customizing the randomized election timeout.

### CODE REVIEW

File: src/raft.rs [hunk 1]
```diff
@@ -124,6 +124,14 @@ pub struct Config {
     /// rejoins the cluster.
     pub pre_vote: bool,
 
+    /// The range of election timeout. In some cases, we hope some nodes has less possibility
+    /// to become leader. This configuration ensures that the randomized election_timeout
+    /// will always be suit in [min_election_tick, max_election_tick).
+    /// If it is None, then election_tick will be chosen.
+    pub min_election_tick: Option<usize>,
```
Comment (siddontang) [inline src/raft.rs]: I think you can remove Option here, you can set min to election timeout and max to 2 x election timeout explicitly if they are 0 at initialization.

**Comment by BusyJay** [comment]
PTAL

File: src/raft.rs [hunk 2]
```diff
@@ -154,6 +180,22 @@ impl Config {
             ));
         }
 
+        let min_timeout = self.min_election_tick();
+        let max_timeout = self.max_election_tick();
+        if min_timeout < self.election_tick {
```
Comment (hicqu) [inline src/raft.rs]: should be `>`?

File: src/raft.rs [hunk 3]
```diff
@@ -154,6 +180,22 @@ impl Config {
...
+                min_timeout, self.election_tick
+            )));
+        }
+
+        if min_timeout >= max_timeout {
+            return Err(Error::ConfigInvalid(format!(
+                "min election tick {} should be less than max election tick {}",
+                min_timeout, max_timeout
+            )));
+        }
```
Comment (hicqu) [inline src/raft.rs]: Don't need to check `max_timeout < election_tick`?

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**